internal/commands/user/list/command_test.go

```diff
@@ -0,0 +1,192 @@
+	t.Run("Should construct a Realm client with the configured base url", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		cmd := &command{inputs: inputs{}}
+		assert.Nil(t, cmd.realmClient)
+
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+	})
+}
+
+func TestUserListHandler(t *testing.T) {
+	testApp := realm.App{
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+
+	newMockClient := func() mock.RealmClient {
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			testApp.GroupID = filter.GroupID
+			testApp.ID = filter.App
+			return []realm.App{testApp}, nil
+		}
+		return realmClient
+	}
+
+	t.Run("Should find app users", func(t *testing.T) {
+		testUsers := []realm.User{{ID: "user1"}, {ID: "user2"}}
+
+		realmClient := newMockClient()
+		realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+			assert.Equal(t, "projectID", groupID)
+			assert.Equal(t, "appID", appID)
+			return testUsers, nil
+		}
+
+		cmd := &command{
+			inputs: inputs{
+				ProjectAppInputs: cli.ProjectAppInputs{
+					Project: "projectID",
+					App:     "appID",
+				},
+			},
+			realmClient: realmClient,
+		}
+
+		assert.Nil(t, cmd.Handler(nil, nil))
+		assert.Equal(t, testUsers, cmd.users)
+	})
+
+	t.Run("Should return an error", func(t *testing.T) {
+		for _, tc := range []struct {
+			description string
+			setupClient func() realm.Client
+			expectedErr error
+		}{
+			{
+				description: "When resolving the app fails",
+				setupClient: func() realm.Client {
+					realmClient := mock.RealmClient{}
+					realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+						return nil, errors.New("something bad happened")
+					}
+					return realmClient
+				},
+				expectedErr: errors.New("something bad happened"),
+			},
+			{
+				description: "When finding the users fails",
+				setupClient: func() realm.Client {
+					realmClient := mock.RealmClient{}
+					realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+						return []realm.App{testApp}, nil
+					}
+					realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+						return nil, errors.New("something bad happened")
+					}
+					return realmClient
+				},
+				expectedErr: errors.New("failed to list users: something bad happened"),
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient := tc.setupClient()
+
+				cmd := &command{
+					realmClient: realmClient,
+				}
+
+				err := cmd.Handler(nil, nil)
+				assert.Equal(t, tc.expectedErr, err)
+			})
+		}
+	})
+}
+
+func TestUserListFeedback(t *testing.T) {
+	for _, tc := range []struct {
+		description    string
+		users          []realm.User
+		expectedOutput string
+	}{
+		{
+			description: "",
```

Comment:
I think you're missing a test case for the empty users case, as well as a proper description here